lib/topologies/server.js

```diff
@@ -21,6 +21,57 @@ var inherits = require('util').inherits,
   SessionMixins = require('./shared').SessionMixins,
   relayEvents = require('./shared').relayEvents;
 
+function getSaslSupportedMechs(options) {
+  if (!options) {
+    return {};
+  }
+
+  const authArray = options.auth || [];
+  const authMechanism = authArray[0] || options.authMechanism;
```

Comment:
This looks like it can still potentially be `undefined`

Reply:
It is possible for it to be `undefined`. That's one of the cases that we want to handle with the `saslSupportedMechs`